replace_with/src/lib.rs

```diff
@@ -0,0 +1,8 @@
+// Julien Chien <jchien17@cmc.edu>
+
+pub fn replace_with<T, F: FnOnce(T) -> T>(t: &mut T, f: F) {
+  unsafe {
+    let mut ptr: *mut T = t;
+    *ptr = f(std::ptr::read(ptr))
```

Comment:
Sorry for the delay.

While cargo reports that both tests are "ok", it also reports that the testing binary produces a SIGSEGV.

As seen in my terminal:
```
$ c test --test required
   Compiling replace_with v0.1.0 (...)
    Finished debug [unoptimized + debuginfo] target(s) in 0.37 secs
     Running target/debug/required-8f822a92fedfe6fe

running 2 tests
test required::complex_type ... ok
test required::simple_type ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured

error: process didn't exit successfully: `/.../target/debug/required-8f822a92fedfe6fe` (signal: 11, SIGSEGV: invalid memory reference)

To learn more, run the command again with --verbose.
```

Does this match what you see?